src/ttsim/interface_dag_elements/fail_if.py

```diff
@@ -322,16 +325,46 @@ def environment_is_invalid(
     policy_environment: PolicyEnvironment,
```

Comment:
Please rename check to `policy_environmen_is_invalid`, I was quite confused when looking at the PR.

Looks like I was not the only one -- please revert the changes to this check, neither the backend nor the evaluation date belongs into the policy environment (see the interface DAG).